stubs/icalendar/icalendar/tools.pyi

```diff
@@ -1,12 +1,23 @@
+import datetime
 from typing import Final
+from typing_extensions import TypeIs
+
+from pytz.tzinfo import BaseTzInfo
 
 from .prop import vText
 
-__all__ = ["UIDGenerator"]
+__all__ = ["UIDGenerator", "is_date", "is_datetime", "to_datetime", "is_pytz", "is_pytz_dt", "normalize_pytz"]
 
 class UIDGenerator:
     chars: Final[list[str]]
     @staticmethod
     def rnd_string(length: int = 16) -> str: ...
     @staticmethod
     def uid(host_name: str = "example.com", unique: str = "") -> vText: ...
+
+def is_date(dt: datetime.date) -> TypeIs[datetime.date]: ...  # and not datetime.date
```

Comment:
```suggestion
def is_date(dt: datetime.date) -> TypeIs[datetime.date]: ...  # and not datetime.datetime
```

Reply:
Also I think using TypeIs is inaccurate here. If it returns TypeIs[date], it should return True for all arguments that are instances of date, which this one doesn't. We could use TypeGuard, but given the parameter annotation, it can't really do useful narrowing anyway, so I'd just use bool as the return type.